SmartDeviceLink/SDLVideoEncoder.m

```diff
@@ -0,0 +1,276 @@
+//
+//  SDLVideoEncoder.m
+//  SmartDeviceLink-iOS
+//
+//  Created by Muller, Alexander (A.) on 12/5/16.
+//  Copyright © 2016 smartdevicelink. All rights reserved.
+//
+
+#import "SDLVideoEncoder.h"
+
+#import "SDLLogMacros.h"
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+NSString *const SDLErrorDomainVideoEncoder = @"com.sdl.videoEncoder";
+static NSDictionary<NSString *, id>* _defaultVideoEncoderSettings;
+
+
+@interface SDLVideoEncoder ()
+
+@property (assign, nonatomic, nullable) VTCompressionSessionRef compressionSession;
+@property (assign, nonatomic, nullable) CFDictionaryRef sdl_pixelBufferOptions;
+@property (assign, nonatomic) NSUInteger currentFrameNumber;
+
+@end
+
+
+@implementation SDLVideoEncoder
+
++ (void)initialize {
+    if (self != [SDLVideoEncoder class]) {
+        return;
+    }
+    
+    _defaultVideoEncoderSettings = @{
+                                     (__bridge NSString *)kVTCompressionPropertyKey_ProfileLevel: (__bridge NSString *)kVTProfileLevel_H264_Baseline_AutoLevel,
+                                     (__bridge NSString *)kVTCompressionPropertyKey_RealTime: @YES
+                                     };
+}
+
+- (instancetype)initWithDimensions:(CGSize)dimensions properties:(NSDictionary<NSString *,id> *)properties delegate:(id<SDLVideoEncoderDelegate> __nullable)delegate error:(NSError * _Nullable __autoreleasing *)error {
+    self = [super init];
+    if (!self) {
+        return nil;
+    }
+    
+    _compressionSession = NULL;
+    _currentFrameNumber = 0;
+    _videoEncoderSettings = properties;
+    
+    _delegate = delegate;
+    
+    OSStatus status;
+    
+    // Create a compression session
+    status = VTCompressionSessionCreate(NULL, dimensions.width, dimensions.height, kCMVideoCodecType_H264, NULL, self.sdl_pixelBufferOptions, NULL, &sdl_videoEncoderOutputCallback, (__bridge void *)self, &_compressionSession);
```

Comment:
The only reason this is "so" much to do is because settings the keys is a pain to actually do. Not sure why would would need to migrate this over as this is required to user the video encoder. `SDLVideoEncoder` will only ever be used internally, not by a developer.